templates/default/engage-nginx-proxy-conf.erb

```diff
@@ -107,9 +107,38 @@ server {
   client_max_body_size 102400m;
   gzip on;
 
-  add_header 'Access-Control-Allow-Origin' '*';
-  add_header 'Access-Control-Allow-Credentials' 'true';
-  add_header 'Access-Control-Allow-Methods' 'GET, OPTIONS';
+  # Avoid the if-is-evil-in-location nginx issue
+  # https://stackoverflow.com/questions/27955233/nginx-config-for-cors-add-header-directive-is-not-allowed
+  # Defaults for
+  # add_header 'Access-Control-Allow-Origin' '*';
+  # add_header 'Access-Control-Allow-Credentials' 'true';
+  # add_header 'Access-Control-Allow-Methods' 'GET, OPTIONS';
+  set $cors_origin "*";
+  set $cors_cred   "true";
+  set $cors_method "GET, OPTIONS";
```

Comment:
Isn't the chat client going to use POST?

Reply:
Actually, it uses PUT. 
The `GET, OPTIONS` defaults listed here are not part of the CORS headers used for [dcex]harvard.edu. 
Look further down, the cors_methods variable is changed to `set $cors_method $http_access_control_request_method;`. This means any method that comes in from [dcex]harvard.edu can get the CORS headers on them. They don't have to be listed. We want PUT and POST to both be available because POST is used for the alia name, if we decide to go with that.